lib/AST/Attr.cpp

```diff
@@ -52,6 +52,1520 @@
+  VisitInheritableAttr(attr);
+}
+
+void AttrVisitor::VisitHLSLSV_DispatchThreadIDAttr(const HLSLSV_DispatchThreadIDAttr &attr) {
+  VisitHLSLAnnotationAttr(attr);
+}
+
+void AttrVisitor::VisitHLSLSV_GroupIndexAttr(const HLSLSV_GroupIndexAttr &attr) {
+  VisitHLSLAnnotationAttr(attr);
+}
+
+void AttrVisitor::VisitHLSLShaderAttr(const HLSLShaderAttr &attr) {
+  VisitInheritableAttr(attr);
+}
+
+void AttrVisitor::VisitHotAttr(const HotAttr &attr) {
+  VisitInheritableAttr(attr);
+}
+
+void AttrVisitor::VisitIBActionAttr(const IBActionAttr &attr) {
+  VisitInheritableAttr(attr);
+}
+
+void AttrVisitor::VisitIBOutletAttr(const IBOutletAttr &attr) {
+  VisitInheritableAttr(attr);
+}
+
+void AttrVisitor::VisitIBOutletCollectionAttr(const IBOutletCollectionAttr &attr) {
+  VisitInheritableAttr(attr);
+}
+
+void AttrVisitor::VisitLikelyAttr(const LikelyAttr &attr) {
+  VisitStmtAttr(attr);
+}
+
+void AttrVisitor::VisitMustTailAttr(const MustTailAttr &attr) {
+  VisitStmtAttr(attr);
+}
+
+void AttrVisitor::VisitNoDerefAttr(const NoDerefAttr &attr) {
+  VisitTypeAttr(attr);
+}
+
+void AttrVisitor::VisitNoInlineAttr(const NoInlineAttr &attr) {
+  VisitDeclOrStmtAttr(attr);
+}
+
+void AttrVisitor::VisitNoMergeAttr(const NoMergeAttr &attr) {
+  VisitDeclOrStmtAttr(attr);
+}
+
+void AttrVisitor::VisitObjCGCAttr(const ObjCGCAttr &attr) {
+  VisitTypeAttr(attr);
+}
+
+void AttrVisitor::VisitObjCInertUnsafeUnretainedAttr(const ObjCInertUnsafeUnretainedAttr &attr) {
+  VisitTypeAttr(attr);
+}
+
+void AttrVisitor::VisitObjCKindOfAttr(const ObjCKindOfAttr &attr) {
+  VisitTypeAttr(attr);
+}
+
+void AttrVisitor::VisitOpenCLConstantAddressSpaceAttr(const OpenCLConstantAddressSpaceAttr &attr) {
+  VisitTypeAttr(attr);
+}
+
+void AttrVisitor::VisitOpenCLGenericAddressSpaceAttr(const OpenCLGenericAddressSpaceAttr &attr) {
+  VisitTypeAttr(attr);
+}
+
+void AttrVisitor::VisitOpenCLGlobalAddressSpaceAttr(const OpenCLGlobalAddressSpaceAttr &attr) {
+  VisitTypeAttr(attr);
+}
+
+void AttrVisitor::VisitOpenCLGlobalDeviceAddressSpaceAttr(const OpenCLGlobalDeviceAddressSpaceAttr &attr) {
+  VisitTypeAttr(attr);
+}
+
+void AttrVisitor::VisitOpenCLGlobalHostAddressSpaceAttr(const OpenCLGlobalHostAddressSpaceAttr &attr) {
+  VisitTypeAttr(attr);
+}
+
+void AttrVisitor::VisitOpenCLLocalAddressSpaceAttr(const OpenCLLocalAddressSpaceAttr &attr) {
+  VisitTypeAttr(attr);
+}
+
+void AttrVisitor::VisitOpenCLPrivateAddressSpaceAttr(const OpenCLPrivateAddressSpaceAttr &attr) {
+  VisitTypeAttr(attr);
+}
+
+void AttrVisitor::VisitOpenCLUnrollHintAttr(const OpenCLUnrollHintAttr &attr) {
+  VisitStmtAttr(attr);
+}
+
+void AttrVisitor::VisitPtr32Attr(const Ptr32Attr &attr) {
+  VisitTypeAttr(attr);
+}
+
+void AttrVisitor::VisitPtr64Attr(const Ptr64Attr &attr) {
+  VisitTypeAttr(attr);
+}
+
+void AttrVisitor::VisitSPtrAttr(const SPtrAttr &attr) {
+  VisitTypeAttr(attr);
+}
+
+void AttrVisitor::VisitAlwaysInlineAttr(const AlwaysInlineAttr &attr) {
+  VisitDeclOrStmtAttr(attr);
```

Comment:
Are there no base-classes that can simplify this?

Reply:
Auto-generated.